com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -336,8 +336,6 @@ void NotifyAgentDone(DoneReason doneReason)
                 UpdateRewardStats();
             }
 
-            // The Agent is done, so we give it a new episode Id
-            m_EpisodeId = EpisodeIdCounter.GetEpisodeId();
```

Comment:
No longer bump the EpisodeId when the Agent is done.
Maybe we should rename the variable.